plugins/in_winevtlog/winevtlog.c

```diff
@@ -0,0 +1,786 @@
+        status = GetLastError();
+        if (ERROR_CANCELLED == status) {
+            return FLB_FALSE;
+        }
+        if (ERROR_NO_MORE_ITEMS != status) {
+            return FLB_FALSE;
+        }
+    }
+
+    if (status == ERROR_SUCCESS) {
+        ch->count = count;
+        for (int i = 0; i < count; i++) {
+            ch->events[i] = hEvents[i];
+            EvtUpdateBookmark(ch->bookmark, ch->events[i]);
+        }
+
+        return FLB_TRUE;
+    }
+
+    return FLB_FALSE;
+}
+
+/*
+ * Read from an open Windows Event Log channel.
+ */
+int winevtlog_read(struct winevtlog_channel *ch, msgpack_packer *mp_pck, struct winevtlog_config *ctx,
+                   unsigned int *read)
+{
+    DWORD status = ERROR_SUCCESS;
+    PWSTR wSystem = NULL;
+    unsigned int system_size = 0, message_size = 0, string_inserts_size = 0;
+    int hit_threshold = FLB_FALSE;
+    unsigned int read_size = 0;
+    PWSTR wMessage = NULL;
+    PEVT_VARIANT pSystem = NULL;
+    PEVT_VARIANT pStringInserts = NULL;
+    UINT countInserts = 0;
+
+    while (winevtlog_next(ch, hit_threshold)) {
+        for (DWORD i = 0; i < ch->count; i++) {
+            if (ctx->render_event_as_xml) {
+                wSystem = render_event(ch->events[i], EvtRenderEventXml, &system_size);
+                wMessage = get_description(ch->events[i], LANG_NEUTRAL, &message_size);
+                get_string_inserts(ch->events[i], &pStringInserts, &countInserts, &string_inserts_size);
+                if (wSystem) {
+                    /* Caluculate total allocated size: system + message + string_inserts */
+                    read_size += (system_size + message_size + string_inserts_size);
+                    winevtlog_pack_xml_event(mp_pck, wSystem, wMessage, pStringInserts, countInserts, ch, ctx);
+
+                    flb_free(pStringInserts);
+                    flb_free(wSystem);
+                    if (wMessage)
+                        flb_free(wMessage);
+                }
+            }
+            else {
+                render_system_event(ch->events[i], &pSystem, &system_size);
+                wMessage = get_description(ch->events[i], LANG_NEUTRAL, &message_size);
+                get_string_inserts(ch->events[i], &pStringInserts, &countInserts, &string_inserts_size);
+                if (pSystem) {
+                    /* Caluculate total allocated size: system + message + string_inserts */
+                    read_size += (system_size + message_size + string_inserts_size);
+                    winevtlog_pack_event(mp_pck, pSystem, wMessage, pStringInserts, countInserts, ch, ctx);
+
+                    flb_free(pStringInserts);
+                    flb_free(pSystem);
+                    if (wMessage)
+                        flb_free(wMessage);
+                }
+            }
+        }
+
+        /* Closes any events in case an error occurred above. */
+        for (DWORD i = 0; i < ch->count; i++) {
```

Comment:
Could you move the definition of this variable to the begining of the function and use a representative name for it?

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/4179/commits/a6583c0c04c813dc128074f8223dd90a3777ce78